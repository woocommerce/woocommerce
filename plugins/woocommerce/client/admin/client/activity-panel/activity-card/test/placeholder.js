/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { createElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { ActivityCardPlaceholder } from '../';

describe( 'ActivityCardPlaceholder', () => {
	test( 'should render a default placeholder', () => {
		const { container } = render( <ActivityCardPlaceholder /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with subtitle placeholder', () => {
		const { container } = render( <ActivityCardPlaceholder hasSubtitle /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with date placeholder', () => {
		const { container } = render( <ActivityCardPlaceholder hasDate /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with action placeholder', () => {
		const { container } = render( <ActivityCardPlaceholder hasAction /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with all optional placeholder', () => {
		const { container } = render(
			<ActivityCardPlaceholder hasAction hasDate hasSubtitle />
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with multiple lines of content', () => {
		const { container } = render( <ActivityCardPlaceholder lines={ 3 } /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with no content', () => {
		const { container } = render( <ActivityCardPlaceholder lines={ 0 } /> );
		expect( container ).toMatchSnapshot();
	} );
} );
