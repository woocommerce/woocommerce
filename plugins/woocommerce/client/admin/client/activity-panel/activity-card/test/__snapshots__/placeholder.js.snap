// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCardPlaceholder should render a card placeholder with action placeholder 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__actions"
    >
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with all optional placeholder 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
      <div
        class="woocommerce-activity-card__subtitle is-placeholder"
      />
      <div
        class="woocommerce-activity-card__date"
      >
        <span
          class="is-placeholder"
        />
      </div>
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__actions"
    >
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with date placeholder 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
      <div
        class="woocommerce-activity-card__date"
      >
        <span
          class="is-placeholder"
        />
      </div>
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with multiple lines of content 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
      <span
        class="is-placeholder"
      />
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with no content 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__body"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with subtitle placeholder 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
      <div
        class="woocommerce-activity-card__subtitle is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a default placeholder 1`] = `
<div>
  <div
    aria-hidden="true"
    class="woocommerce-activity-card is-loading"
  >
    <span
      class="woocommerce-activity-card__icon"
    >
      <span
        class="is-placeholder"
      />
    </span>
    <div
      class="woocommerce-activity-card__header"
    >
      <div
        class="woocommerce-activity-card__title is-placeholder"
      />
    </div>
    <div
      class="woocommerce-activity-card__body"
    >
      <span
        class="is-placeholder"
      />
    </div>
  </div>
</div>
`;
